### ISSUE TITLE
Reenable unit tests compilation and execution on all CI platforms.

### ISSUE THREAD

**Issue by CusiniM** [GEOS-DEV/GEOS#2512]
In order to stay within the 3hrs limit on Travis, most gpu jobs do not run nor compile the unit tests. Once we move to github actions we should reenable them for all CI jobs.

**Comment by wrtobin** [comment]
I can put together a PR for this, as I also want to get started on adding a CI that covers the HIP/ROCM build.

@bmhan12 Can you give me any insight into the dev workflow to put together the CI docker environments? I know AMD provides HIP/ROCM containers so it should be doable.

**Comment by bmhan12** [comment]
> Can you give me any insight into the dev workflow to put together the CI docker environments? I know AMD provides HIP/ROCM containers so it should be doable without too much difficulty beyond building our TPLs I would think.

If it's anything like nvidia, they have pre-built images with CUDA installed already:
https://hub.docker.com/r/nvidia/cuda

Then it's a matter of building our dependencies on top of the pre-built images:
https://github.com/GEOS-DEV/thirdPartyLibs/blob/master/docker/clang-cuda/Dockerfile

**Comment by wrtobin** [comment]
Did our travis platforms have physical GPUs?

**Comment by CusiniM** [comment]
> Did our travis platforms have physical GPUs?

No, they didn't. I think that's why we were only building the tests and not executing them. I guess we can just re-enable the building of the tests for now and add the execution once we have the local machine ready.

**Comment by paveltomin** [comment]
@TotoGaz was this resolved after moving into GHA?
